test_communication/test/test_secure_publisher.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright 2015-2017 Open Source Robotics Foundation, Inc.
```

Comment:
this file is very similar to test_publisher.cpp in this package. I minimized the change but didn't want to make the change in the original file because I'm relying on hardcoded "fixed" node names for the security test and propagating this change to the regular test_communication tests would allow possible cross talk from one test to another and increase flakyness. In this context it's ok because we test only for Fast-RTPS that doesn't take long to tear down nodes so crosstalk should not happen